init.lua

```diff
@@ -174,19 +174,55 @@ local function update_entity(entity)
 				end
 			end
 		end
+	end
+	if active_lights[pos_str] then
 		minetest.get_node_timer(pos):start(cleanup_interval)
 	end
 	entity.update = false
 end
 
+
+-- Save the original nodes timer if it has one
+local function save_timer(pos)
+	-- Convert the position back to a vector
+	local pos_vec = minetest.string_to_pos(pos)
+	local timer=minetest.get_node_timer(pos_vec)
+	if timer:is_started() then
+		local meta = minetest.get_meta(pos_vec)
+		meta:set_float("saved_timer_timeout", timer:get_timeout())
+		meta:set_float("saved_timer_elapsed", timer:get_elapsed())
+	end
+end
+
+-- Restore the original nodes timer if it had one
+local function restore_timer(pos_vec)
+	-- Convert the position back to a vector
+--	local pos_vec = minetest.string_to_pos(pos)
+	local meta = minetest.get_meta(pos_vec)
+	local timeout = meta:get_float("saved_timer_timeout")
+	if timeout then
+		local elapsed = meta:get_float("saved_timer_elapsed")
+		local timer=minetest.get_node_timer(pos_vec)
+		timer:set(timeout,elapsed)
+		meta:set_string("saved_timer_timeout",nil)
+		meta:set_string("saved_timer_elapsed",nil)
```

Comment:
Invalid API call. Use `""` to remove the metadata entry.

Reply:
oh thanks.  wasn't sure on that.  

fyi, we discovered another bug with the saved timers so will resubmit this pull request once we've confirmed it's fixed

Reply:
We found the saved timer was being overwritten with the light nodes timer so we added a check to make sure we're not saving light node timers.  I'll submit a pull request with the adjustments after the weekend.

Reply:
@Izzyb1234 Did you yet have the chance to work on this?

Reply:
@SmallJoker Sorry got distracted with IRL. I think we have it working well in Exile now so I'll see about extracting the changes since this PR was first submitted.

Reply:
Ok, I believe all the changes have been committed now. Let me know if there are any issues.